### PR TITLE
Master 1039764

### PR DESCRIPTION
**Pull request by noelpower**

### CODE REVIEW

File: src/modules/SambaNetJoin.pm [hunk 1]
```diff
@@ -231,31 +231,35 @@ sub Test {
     my $netbios_name 	= SambaConfig->GlobalGetStr("netbios name", undef);
     my $conf_file	= SCR->Read (".target.tmpdir")."/smb.conf";
     my $include		= "";
```
Comment (ddiss) [inline src/modules/SambaNetJoin.pm]: IIUC, the two $include variables in this file can now be dropped?

File: src/modules/SambaNetJoin.pm [hunk 2]
```diff
@@ -302,34 +309,33 @@ sub Join {
...
         }
-	SCR->Write (".target.string", $conf_file, $content);
 	$cmd		= "KRB5_CONFIG=$krb_file ";
 	SCR->Write (".target.string", $krb_file, "[realms]\n\t$realm = {\n\tkdc = $server\n\t}\n");
     }
     else {
-	SCR->Write (".target.string", $conf_file, "[global]$include\n\tsecurity = domain\n\tworkgroup = $domain\n");
+	$glb_overrides{"realm"} = undef;
+	$glb_overrides{"workgroup"} = $domain;
+	$glb_overrides{"security"} = "domain";
```
Comment (ddiss) [inline src/modules/SambaNetJoin.pm]: How did the previous implementation handle leaving an AD domain, if at all. Is this behaviour affected by this change?

File: src/modules/SambaNetJoin.pm (hunk 1 above)
Comment (noelpower) [inline src/modules/SambaNetJoin.pm]: yup, guess so, these are no longer used

File: src/modules/SambaNetJoin.pm (hunk 2 above)
Comment (noelpower) [inline src/modules/SambaNetJoin.pm]: no, leaving assumes everything (including the smb.conf) is already setup in place etc., I will force a new branch here with the includes removed

**Pull request merged**